File: Pertemuan-13/Tugas/AppPerpus/Pengembalian.py
import requests
import json
class Pengembalian:
    def __init__(self):
        self.__id=None
        self.__kode = None
        self.__tglkembali = None
        self.__denda = None
        self.__kodebuku = None
        self.__niap = None
        self.__nipp = None
        self.__url = "http://f0832375.xsph.ru/app/pengembalian_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def kode(self):
        return self.__kode
        
    @kode.setter
    def kode(self, value):
        self.__kode = value
    @property
    def tglkembali(self):
        return self.__tglkembali
        
    @tglkembali.setter
    def tglkembali(self, value):
        self.__tglkembali = value
    @property
    def denda(self):
        return self.__denda
        
    @denda.setter
    def denda(self, value):
        self.__denda = value
    @property
    def kodebuku(self):
        return self.__kodebuku
        
    @kodebuku.setter
    def kodebuku(self, value):
        self.__kodebuku = value
    @property
    def niap(self):
        return self.__niap
        
    @niap.setter
    def niap(self, value):
        self.__niap = value
    @property
    def nipp(self):
        return self.__nipp
        
    @nipp.setter
    def nipp(self, value):
        self.__nipp = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_kode(self, kode):
        url = self.__url+"?kode="+kode
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id']
            self.__kode = item['kode']
            self.__tglkembali = item['tglkembali']
            self.__denda = item['denda']
            self.__kodebuku = item['kodebuku']
            self.__niap = item['niap']
            self.__nipp = item['nipp']
        return data
    def simpan(self):
        payload = {
            "kode":self.__kode,
            "tglkembali":self.__tglkembali,
            "denda":self.__denda,
            "kodebuku":self.__kodebuku,
            "niap":self.__niap,
            "nipp":self.__nipp
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_kode(self, kode):
        url = self.__url+"?kode="+kode
        payload = {
            "kode":self.__kode,
            "tglkembali":self.__tglkembali,
            "denda":self.__denda,
            "kodebuku":self.__kodebuku,
            "niap":self.__niap,
            "nipp":self.__nipp
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_kode(self,kode):
        url = self.__url+"?kode="+kode
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text

File: Pertemuan-12/AppPerpus/Pengembalian.php
<?php
//Simpanlah dengan nama file : Pengembalian.php
require_once 'database.php';
class Pengembalian 
{
    private $db;
    private $table = 'pengembalian';
    public $kode = "";
    public $tglkembali = "";
    public $denda = "";
    public $kodebuku = "";
    public $niap = "";
    public $nipp = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_kode(int $kode)
    {
        $query = "SELECT * FROM $this->table WHERE kode = $kode";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`kode`,`tglkembali`,`denda`,`kodebuku`,`niap`,`nipp`) VALUES ('$this->kode','$this->tglkembali','$this->denda','$this->kodebuku','$this->niap','$this->nipp')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET kode = '$this->kode', tglkembali = '$this->tglkembali', denda = '$this->denda', kodebuku = '$this->kodebuku', niap = '$this->niap', nipp = '$this->nipp' 
        WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_kode($kode): int
    {
        $query = "UPDATE $this->table SET kode = '$this->kode', tglkembali = '$this->tglkembali', denda = '$this->denda', kodebuku = '$this->kodebuku', niap = '$this->niap', nipp = '$this->nipp' 
        WHERE kode = $kode";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_kode($kode): int
    {
        $query = "DELETE FROM $this->table WHERE kode = $kode";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>